lib/wasi/src/state/builder.rs

```diff
@@ -508,17 +490,21 @@ mod test {
 
     #[test]
     fn env_var_errors() {
+        // `a=b` means key is `a` and value is `b`, which is OK.
+        // `a=b=c` means key is `a` and value is `b=c`, which is OK too.
         let output = create_wasi_state("test_prog")
-            .env("HOM=E", "/home/home")
```

Comment:
Do we want to put the test checking the failure of `=` in the key back?

Reply:
Yes. I've updated the test cases to handle all the scenarios.